objects/src/accounts/vault.rs

```diff
@@ -0,0 +1,68 @@
+use super::{AccountError, AccountId, Asset, Digest};
+
+// ACCOUNT VAULT
+// ================================================================================================
+
+/// An asset container for an account.
+///
+/// An account vault can contain an unlimited number of assets. The assets are stored in a Sparse
```

Comment:
Question: why is there a limit for the note input but not for the acocount? https://github.com/0xPolygonMiden/miden-base/issues/10

Reply:
Fo notes, we need to have a cap so that we can lay them out predictably in memory without introducing too much overhead. So, if we know that a note can contain at most 1000 assets, we can space multiple notes out accordingly.

Basically, we need to set a limit somewhere - it could be fairly large, but I think there is an argument for making it relatively small (i.e., smaller notes introduce fewer potential attack vectors). Moreover, I don't think this is too constraining as use cases for notes carrying thousands of assets are probably few.

For accounts, we don't want to set a limit as it is difficult to predict how many assets a given account could hold. I can easily imagine use cases where a single account could hold many millions of assets.